src/redux/bidData/index.ts

```diff
@@ -39,7 +48,7 @@ export const initialBidRequest = (payload: boolean) => ({
   type: ActionTypes.INITIAL_BID_REQUEST,
 })
 
-export const initialBidSuccess = (payload: AuctionBid[]) => ({
+export const initialBidSuccess = (payload: BidsBySaleId) => ({
   payload,
   type: ActionTypes.INITIAL_BID_SUCCESS,
 })
```

Comment:
This action expects that the line calling it to know about and pass previous Redux state (precisely `store.bidData.bids`). Actions shouldn't accept previous data from the redux.

Reply:
I can only call this action when I make a request to the subgraph and so I don't pass any previous states into this action. Its only called by this function 

```typescript

export const fetchSaleBids = (id: string, saleType: saleType, saleBidsRequest: Promise<any>): AppThunk => {
  return async dispatch => {
    dispatch(initialBidRequest(true))
    try {
      dispatch(initialBidSuccess(await generateInitialSaleData(saleBidsRequest, saleType)))
    } catch (error) {
      console.log(error)
      dispatch(initialBidFailure(error))
    }
  }
}
```

Can you elaborate more on what the optimal way to write this piece of code in?